spec/Section 6 -- Execution.md

```diff
@@ -660,7 +666,8 @@ ResolveFieldValue(objectType, objectValue, fieldName, argumentValues):
 Note: It is common for {resolver} to be asynchronous due to relying on reading
 an underlying database or networked service to produce a value. This
 necessitates the rest of a GraphQL executor to handle an asynchronous execution
-flow.
+flow. In addition, an implementation for collections may leverage asynchronous
+iterators or asynchronous generators provided by many programming languages.
```

Comment:
This would introduce the term "generator" and "iterator" to the spec, which isn't currently used. Let's keep the language simpler:

```suggestion
flow. In addition, for fields that have a return type that is a List type, each
value in a collection of values returned by {resolver} may itself be
asynchronous.
```

Reply:
What about something very similar:

```suggestion
flow. In addition, for fields that have a return type that is a List type, each
value in a collection of values returned by {resolver} may be
retrieved asynchronously.
```

This just avoids saying that the value itself is asynchronous, which I think you might mean to just be a more concise version of the above?

@robrichard may have input, this was extracted from #742